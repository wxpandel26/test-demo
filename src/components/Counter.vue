<template>
    <div>
        <span>count</span>
        <span>ttt</span>
        <button @click="handlerClick">count++</button>
    </div>
</template>

<script>
export default {
    data() {
        return {
            count : 0
        }
    },
    methods :{
        handlerClick() {
            this.count++;
            this.$emit("change",this.count);
        }
    }
}
</script>

<style>

</style>
