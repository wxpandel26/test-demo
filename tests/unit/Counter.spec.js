import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter.vue", () => {
  const wrapper = mount(Counter);
  it("renders props.msg when passed", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("count++", () => {
    const button = wrapper.find("button");
    button.trigger("click");
    expect(wrapper.vm.count).toBe(2);
  });
});
