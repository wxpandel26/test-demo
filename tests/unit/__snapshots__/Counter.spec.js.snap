// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders props.msg when passed 1`] = `<div><span>count</span> <button>count++</button></div>`;
